pkg/sql/importer/exportparquet_test.go

```diff
@@ -143,84 +144,130 @@ func validateParquetFile(
 	sqlDB.Exec(t, fmt.Sprintf("CREATE DATABASE %s", dbName))
 
-	tableName := "table_1"
-	// TODO (butler): Randomly generate additional table(s) using tools from PR
-	// #75677. The function below only creates a deterministic table with a bunch
-	// of interesting corner case values.
-	err := randgen.GenerateRandInterestingTable(tc.Conns[0], dbName, tableName)
-	require.NoError(t, err)
-
-	s0 := tc.Server(0)
-	ie := s0.ExecutorConfig().(sql.ExecutorConfig).InternalExecutor
+	var tableName string
+	ie := srv.ExecutorConfig().(sql.ExecutorConfig).InternalExecutor
+	// Try at most 10 times to populate a random table with at least 10 rows.
 	{
-		// Ensure table only contains columns supported by EXPORT Parquet
-		_, cols, err := ie.QueryRowExWithCols(
-			ctx,
-			"",
-			nil,
-			sessiondata.InternalExecutorOverride{
-				User:     security.RootUserName(),
-				Database: dbName,
-			},
-			fmt.Sprintf("SELECT * FROM %s LIMIT 1", tableName))
-		require.NoError(t, err)
-
-		for _, col := range cols {
-			_, err := importer.NewParquetColumn(col.Typ, "", false)
-			if err != nil {
-				t.Logf("Column type %s not supported in parquet, dropping", col.Typ.String())
-				sqlDB.Exec(t, fmt.Sprintf(`ALTER TABLE %s DROP COLUMN %s`, tableName, col.Name))
+		var (
+			i           int
+			success     bool
+			tablePrefix = "table"
+			numTables   = 20
+		)
+
+		stmts := randgen.RandCreateTables(rng, tablePrefix, numTables,
+			randgen.PartialIndexMutator,
+			randgen.ForeignKeyMutator,
+		)
+
+		var sb strings.Builder
+		for _, stmt := range stmts {
+			sb.WriteString(tree.SerializeForDisplay(stmt))
+			sb.WriteString(";\n")
+		}
+		sqlDB.Exec(t, sb.String())
+
+		for i = 1; i < numTables; i++ {
+			tableName = tablePrefix + fmt.Sprint(i)
+			numRows, err := randgen.PopulateTableWithRandData(rng, db, tableName, 20)
+			require.NoError(t, err)
+			if numRows > 5 {
+				// Ensure the table only contains columns supported by EXPORT Parquet. If an
+				// unsupported column cannot be dropped, try populating another table
+				if err := func() error {
+					_, cols, err := ie.QueryRowExWithCols(
+						ctx,
+						"",
+						nil,
+						sessiondata.InternalExecutorOverride{
+							User:     security.RootUserName(),
+							Database: dbName},
+						fmt.Sprintf("SELECT * FROM %s LIMIT 1", tableName))
+					require.NoError(t, err)
+
+					for _, col := range cols {
+						_, err := importer.NewParquetColumn(col.Typ, "", false)
+						if err != nil {
+							_, err = sqlDB.DB.ExecContext(ctx, fmt.Sprintf(`ALTER TABLE %s DROP COLUMN %s`, tableName, col.Name))
+							if err != nil {
+								return err
+							}
+						}
+					}
+					return nil
+				}(); err != nil {
+					continue
+				}
+				success = true
+				break
 			}
 		}
+		require.Equal(t, true, success, "test flake: failed to create a random table")
 	}
-
-	// TODO (butler): iterate over random select statements
+	t.Logf("%s", sqlDB.QueryStr(t, `SHOW CREATE TABLE `+tableName))
```

Comment:
[nit] Since I imagine we are only reading this on a failure, perhaps add some context here to help whoever is reading through the log: 
```suggestion
	t.Logf("exporting as parquet from random table with schema: %s", sqlDB.QueryStr(t, `SHOW CREATE TABLE `+tableName))
```